spark3/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -56,48 +83,85 @@ public boolean supportsExternalMetadata() {
   }
 
   @Override
-  public SparkTable getTable(StructType schema, Transform[] partitioning, Map<String, String> options) {
-    // Get Iceberg table from options
-    Configuration conf = SparkSession.active().sessionState().newHadoopConf();
-    Table icebergTable = getTableAndResolveHadoopConfiguration(options, conf);
-
-    // Build Spark table based on Iceberg table, and return it
-    // Eagerly refresh the table before reading to ensure views containing this table show up-to-date data
-    return new SparkTable(icebergTable, schema, true);
+  public Table getTable(StructType schema, Transform[] partitioning, Map<String, String> options) {
+    String catalogName = extractCatalog(new CaseInsensitiveStringMap(options));
+    Identifier ident = extractIdentifier(new CaseInsensitiveStringMap(options));
+    CatalogManager catalogManager = SparkSession.active().sessionState().catalogManager();
+    CatalogPlugin catalog = catalogManager.catalog(catalogName);
+    try {
+      if (catalog instanceof TableCatalog) {
+        return ((TableCatalog) catalog).loadTable(ident);
+      }
+    } catch (NoSuchTableException e) {
+      // throwing an iceberg NoSuchTableException because the Spark one is typed and cant be thrown from this interface
+      throw new org.apache.iceberg.exceptions.NoSuchTableException(e, "Cannot find table for %s.", ident);
+    }
+    // throwing an iceberg NoSuchTableException because the Spark one is typed and cant be thrown from this interface
+    throw new org.apache.iceberg.exceptions.NoSuchTableException("Cannot find table for %s.", ident);
   }
 
-  protected Table findTable(Map<String, String> options, Configuration conf) {
+  private Spark3Util.CatalogAndIdentifier catalogAndIdentifier(CaseInsensitiveStringMap options) {
     Preconditions.checkArgument(options.containsKey("path"), "Cannot open table: path is not set");
     String path = options.get("path");
-
-    if (path.contains("/")) {
-      HadoopTables tables = new HadoopTables(conf);
-      return tables.load(path);
+    SparkSession spark = SparkSession.active();
+    Spark3Util.CatalogAndIdentifier catalogAndIdentifier;
+    try {
+      catalogAndIdentifier = Spark3Util.catalogAndIdentifier(spark, path);
```

Comment:
Is this try/catch to avoid the `/` check? I think that's a pretty reasonable check. I'm not sure that this is a good idea because the cases you listed above would pass this.

For example, this uses a valid identifier, but is clearly a path reference:
```java
spark.format("iceberg").load("catalog.`file:/path/to/table`");
```

As I mentioned above, I don't think that we should support the mixed path and catalog identifiers, but I still think it is probably a more predictable check to just look for `/`.

Reply:
the try component takes care of `catalog.table`, `catalog.namespace.table` and ```catalog.`file://path/to/table` ``` but doens't take care of `file://path/to/table` which throws a parse exception. The catch part deals w/ `file://path/to/table`.

I suppose we can check 
* if there is a ``` ` ``` character and a `/` and reject (don't mix catalogs and paths)
* If just `/` then treat as path
* assume its tables and parse it

Reply:
Yeah, I think that the problem is that ```catalog.`file://path/to/table` ``` is ambiguous and should not be supported. But parsing it successfully means that it would be translated to a path-based table with a catalog.

I don't think a more complicated check is needed, to see if the `/` is probably escaped. If there is a `/`, then it's a path. That's a really simple rule to follow.

Reply:
After thnking more I have thrown out ``` catalog.`file:/path/to/table` ``` and simplified the method